dragonfly_doe2/doe/floor_space.py

```diff
@@ -0,0 +1,149 @@
+from dragonfly.room2d import Room2D
+from dragonfly.story import Story
+from dataclasses import dataclass
+from typing import List
+
+
+@dataclass
+class Wall:
+    """*.inp Wall object.
+    Example:
+        .. code-block::f#
+            "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+               CONSTRUCTION    = "EWall Construction"
+               LOCATION        = SPACE-V1
+               ..
+    """
+    name: str = None
+    location_index: int = None
+    wall_con_name: str = None
+
+    @classmethod
+    def from_room_seg(cls, _name: str, _bc_id: int, _wall_con_name: str):
+        indexed_id = _bc_id+1
+        return cls(str(_name+f'_Wall_{indexed_id}'), indexed_id, _wall_con_name)
+
+    def to_inp(self):
+        return f'"{self.name}" = EXTERIOR-WALL\n' \
+            f'   CONSTRUCTION   = "{self.wall_con_name}"\n' \
+            f'   LOCATION       = SPACE-V{self.location_index}\n   ..'
+
+    def __repr__(self) -> str:
+        return self.to_inp()
+
+
+@dataclass
+class Space:
+    """The Space Object.
+    Each Room2D has a Space obj. This obj contains windows, walls, doors
+    data.
+    .. code-block::f#
+        "simple_example_dfb_Floor1_Room1" = SPACE
+            SHAPE            = POLYGON
+            POLYGON          = "simple_example_dfb_Floor1_Room1 Plg"
+            C-ACTIVITY-DESC  = *Generic Office Program*
+            ..
+        "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+            CONSTRUCTION    = "EWall Construction"
+            LOCATION        = SPACE-V1
+            ..
+        "simple_example_dfb_Floor1_Room1 Wall_2" = EXTERIOR-WALL
+            CONSTRUCTION    = "EWall Construction"
+            LOCATION        = SPACE-V2
+            ..
+    """
+    name: str = None
+    activity: str = None
+    walls: List[Wall] = None
+
+    @classmethod
+    def from_room(cls, room: Room2D):
+        if not isinstance(room, Room2D):
+            # TODO: Make raise exception rather than print
+            print(
+                f'Unsupported Type: {type(room)}.\n'
+                'Expected dragonfly.room2d.Room2D'
+            )
```

Comment:
Why not do it now? We should only support Room2D, right?

```suggestion
            # TODO: Make raise exception rather than print
            raise ValueError(
                f'Unsupported Type: {type(room)}.\n'
                'Expected dragonfly.room2d.Room2D'
            )
```